src/supervised/distance.jl

```diff
@@ -108,6 +108,40 @@ islipschitzcont_deriv(::L2DistLoss) = true
 isconvex(::L2DistLoss) = true
 isstronglyconvex(::L2DistLoss) = true
 
+
+# ===========================================================
+# L(y, t) = 1 - cos((y-t)*2π/c)
+
+"""
+`PeriodicLoss <: DistanceLoss`
+
+Measures distance on a circle of specified circumference.
+""" 
+immutable PeriodicLoss{T<:AbstractFloat} <: DistanceLoss
+    k::T   # k = 2π/circumference
+    PeriodicLoss(circ::T) = new(T(2π/circ))
```

Comment:
Then again I suppose if one specifies a zero an exception is thrown anyway. so maybe this is actually fine as it is (if a negative `circ` is allows)